app/src/main/java/org/asteroidos/sync/asteroid/IAsteroidDevice.java

```diff
@@ -0,0 +1,43 @@
+package org.asteroidos.sync.asteroid;
+
+import org.asteroidos.sync.connectivity.IConnectivityService;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface IAsteroidDevice {
+    String name = "";
+    String macAddress = "";
+    int batteryPercentage = 0;
+    boolean bonded = false;
+
+    enum ConnectionState {
+        STATUS_CONNECTED,
+        STATUS_CONNECTING,
+        STATUS_DISCONNECTED
+    }
+
+    /**
+     * Can be used to determine if an {@link IAsteroidDevice} is {@link IAsteroidDevice.ConnectionState#STATUS_CONNECTED},
+     * {@link IAsteroidDevice.ConnectionState#STATUS_CONNECTING} or {@link IAsteroidDevice.ConnectionState#STATUS_DISCONNECTED}.
+     *
+     * <pre>
+     * &#064;Override
+     * <b>public ConnectionState</b> getConnectionState() {
+     * //determine connection state
+     * //return {@link ConnectionState}
+     * }
+     * </pre>
+     *
+     * @return Current state of the {@link IAsteroidDevice} connection.
+     */
+    ConnectionState getConnectionState();
+
+    void sendToDevice(UUID characteristic, byte[] data, IConnectivityService service);
```

Comment:
All call places look like `mDevice.sendToDevice`. I don't think it's useful to repeat `sendToDevice`. It should be pretty clear with `send` already.